src/os_macosx.m

```diff
@@ -421,18 +418,14 @@ - (void) sound:(NSSound *)sound didFinishPlaying:(BOOL)flag
 	    delete_sound_callback(_callback);
 	    _callback = NULL;
 	}
-	[sounds_list removeObjectForKey:[NSNumber numberWithLong:_sound_id]];
```

Comment:
Isn't this going to leak? You moved this to `sound_mch_stop()` but that function is only called if you manually stop a playing sound. This delegate is going to called when you do that, but it's **also** going to be called when the sound finishes playing naturally. This function was placed here for that reason.